executor.md

```diff
@@ -0,0 +1,24 @@
+# The `executor` Merkle leaf field
```

Comment:
@RPate97 lmk your thoughts

Reply:
The issue you bring up is legitimate, but I wouldn't solve it by removing the `executor` field. That field gives us a lot of flexibility to implement various schemes for selecting the executor without changing the module. I want to preserve that optionality. 

I would go with one of these two options to resolve the issue instead:
1. Update the `ManagedService` contract to be permissionless. This is functionality equivalent to having no specific `executor` but preserves the field. 
2. Do a better job of securing the `ManagedService` contract. This might involve reevaluating how it is deployed so that it is owned by a multisig wallet as soon as it is deployed. 

Both options have pros and cons. I'm happy to hop on a call to discuss further if you want.

Reply:
> That field gives us a lot of flexibility to implement various schemes for selecting the executor without changing the module. I want to preserve that optionality.

Could you expand on this? I don't see the benefit of keeping this field with our current system

Reply:
I guess this field would be useful if the user wants to be the only one that can execute the deployment? This has never been requested before, and would require us to build additional tooling, but I can understand why a user would want to do this in theory

Reply:
Actually, I remember Kelvin mentioning a long time ago that Optimism would probably want to execute their own deployments because they didn't want to rely on a remote executor that wasn't battle-tested. I'm not sure whether they would've cared if execution was permissionless though. I think teams with higher security requirements could reasonably want to be the sole executor

Reply:
The benefit is that it allows us to implement an unlimited number of different ways to determine who can execute the deployment without needing to change the module. Any future feature we implement related to this will be fully backwards compatible.

#### Non-exhaustive list of examples
1. Fully permissionless (this is what you are suggesting anyone can execute the deployment)
2. Fully permissioned (only we can execute the deployment)
3. Decentralized (the executor is selected through some decentralized process like a fee auction or staking)
4. Only the user (we use a specific address owned by the user)

If we remove the `executor` field entirely, we commit fully to option 1. If we ever want to change this, we must update the module. I don't feel great about this because I don't have evidence to indicate that we won't want to do something else in the future. 

So, since we can achieve something functionally equivalent to option 1 while preserving the `executor` field, I would do that to maintain optionality.

Reply:
It's worth noting that option 2 is probably a lot better from a business perspective.

Reply:
See telegram if you want to talk about this more. I would prefer to hop on a call.